### PR TITLE
Update protobuf to 3.20.0

### PR DESCRIPTION
**Pull request by thii**
And add prebuilt protoc binaries for Apple silicon.

### CODE REVIEW

**Comment by comius** [comment]
I don't think we should be updating precompiled protobuf to rc versions.

**Comment by thii** [comment]
I'll update this when 3.20 is released.

**Comment by comius** [comment]
3.20 was released.

**Comment by thii** [comment]
Updated!

File: proto/private/dependencies.bzl [hunk 1]
```diff
@@ -113,6 +121,14 @@ dependencies = {
         ],
     },
     # Dependency of `com_github_protocolbuffers_protobuf`.
+    "rules_pkg": {
+        "sha256": "a89e203d3cf264e564fcb96b6e06dd70bc0557356eb48400ce4b5d97c2c3720d",
+        "urls": [
+            "https://mirror.bazel.build/github.com/bazelbuild/rules_pkg/releases/download/0.5.1/rules_pkg-0.5.1.tar.gz",
+            "https://github.com/bazelbuild/rules_pkg/releases/download/0.5.1/rules_pkg-0.5.1.tar.gz",
+        ],
+    },
```
Comment (comius) [inline proto/private/dependencies.bzl:130]: This is a new dependency for com_github_protocolbuffers_protobuf?

**Review by comius — APPROVED** [review]

File: proto/private/dependencies.bzl (hunk 1 above)
Comment (thii) [reply proto/private/dependencies.bzl:130]: Yes, without it, the build would error out https://buildkite.com/bazel/rules-proto/builds/1165#f9f1919d-b4ab-4234-bdb5-1382c077d0d3

File: proto/private/dependencies.bzl [hunk 2]
```diff
@@ -23,75 +23,83 @@ dependencies = {
...
+            "https://mirror.bazel.build/github.com/protocolbuffers/protobuf/releases/download/v3.20.0/protoc-3.20.0-linux-x86_64.zip",
+            "https://github.com/protocolbuffers/protobuf/releases/download/v3.20.0/protoc-3.20.0-linux-x86_64.zip",
+        ],
+    },
+    "com_google_protobuf_protoc_macos_aarch64": {
+        "build_file": "@rules_proto//proto/private:BUILD.protoc",
+        "sha256": "0019dfc4b32d63c1392aa264aed2253c1e0c2fb09216f8e2cc269bbfb8bb49b5",
+        "urls": [
+            "https://mirror.bazel.build/github.com/protocolbuffers/protobuf/releases/download/v3.20.0/protoc-3.20.0-osx-aarch64.zip",
+            "https://github.com/protocolbuffers/protobuf/releases/download/v3.20.0/protoc-3.20.0-osx-aarch64.zip",
```
Comment (comius) [inline proto/private/dependencies.bzl:78]: File not found

Comment (comius) [reply proto/private/dependencies.bzl:78]: It seems there's an underscore missing:
`https://github.com/protocolbuffers/protobuf/releases/download/v3.20.0/protoc-3.20.0-osx-aarch_64.zip`

**Review by comius — CHANGES_REQUESTED** [review]
Please fix https://mirror.bazel.build/github.com/protocolbuffers/protobuf/releases/download/v3.20.0/protoc-3.20.0-osx-aarch_64.zip

**Comment by comius** [comment]
I mirrored the binaries.

**Review by comius — APPROVED** [review]

**Pull request merged**